File: Problem108/Solution.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func sortedArrayToBST(nums []int) *TreeNode {
	if nums == nil || len(nums) == 0 {
		return nil
	}
	return getBST(nums, 0, len(nums)-1)
}

func getBST(nums []int, start int, end int) *TreeNode {
	if start > end {
		return nil
	}
	mid := start + (end-start)/2
	root := TreeNode{Val: nums[mid]}
	root.Left = getBST(nums, start, mid-1)
	root.Right = getBST(nums, mid+1, end)
	return &root
}

File: Problem108/Solution.java
/**
 * @author tgorthi
 * @since Jun 2020
 */
public class Solution
{
    public TreeNode sortedArrayToBST(int[] nums)
    {
        if (nums == null || nums.length == 0)
        {
            return null;
        }
        return getBST(nums, 0, nums.length - 1);
    }

    private TreeNode getBST(int[] nums, int start, int end)
    {
        if (start > end)
        {
            return null;
        }
        int mid = start + (end - start) / 2;
        TreeNode left = getBST(nums, start, mid - 1);
        TreeNode right = getBST(nums, mid + 1, end);
        TreeNode root = new TreeNode(nums[mid]);
        root.left = left;
        root.right = right;
        return root;
    }
}